app/script/assets/AssetMapper.js

```diff
@@ -23,31 +23,57 @@ window.z = window.z || {};
 window.z.assets = z.assets || {};
 
 z.assets.AssetMapper = {
-  mapProfileAssets: (entity, assets) => {
-    return assets.filter(asset => asset.type === 'image').map(asset => {
+  /**
+   * Creates asset entities out of raw asset data.
+   *
+   * @param {string} userId - the id of the user that will hold the asset
+   * @param {Array<Object>} assets - the assets to map
+   * @returns {MappedAssets} Object containing the mapped assets
+   */
+  mapProfileAssets: (userId, assets) => {
```

Comment:
I needed to split the mapper's functions in two pure functions that do the mapping and a non pure function that actually put the result of the mapping into a user entity.

This way we have exactly the same signature for both methods, and they can be used without a user entity lying around